js/alert.js

```diff
@@ -0,0 +1,177 @@
+let IsGettingAlertsucceed = false;
+
+function alertHandler() {
+  getAlert();
+  setAlertTable(IsGettingAlertsucceed);
+  setAlertInterval(IsGettingAlertsucceed);
+}
+
+function getAlert() {
+  const alertRateValue = Number(alertRate.value);
+
+  // change IsGettingAlertsucceed value to false if it is true from previous time
+  IsGettingAlertsucceed = false;
+
+  let isItNewAlert = true;
+  let alertIndex;
+  alerts.forEach((alertItem, index) => {
+    if (
+      alertItem.base === alertBaseList.value &&
+      alertItem.target === alertTargetList.value
+    ) {
+      isItNewAlert = false;
+      alertIndex = index;
+    }
+  });
+
+  if (!alertBaseList.value) {
+    showErrorMessage(alertBaseList, alertError, "Select A Currency Please");
+  } else if (!alertTargetList.value) {
+    showErrorMessage(alertTargetList, alertError, "Select A Currency Please");
+  } else if (alertRateValue <= 0) {
+    showErrorMessage(alertRate, alertError, "Invalid Input");
+  } else if (!isItNewAlert) {
+    setAlertYesNoQuestionBox(
+      alertIndex,
+      alertRateValue,
+      "An Alert Has Already Been Set For This Rate. Would You Like To Update It?!"
+    );
+  } else {
+    alerts.push({
+      id: `${alertBaseList.value}-${alertTargetList.value}`,
+      base: alertBaseList.value,
+      target: alertTargetList.value,
+      rate: alertRateValue,
+    });
+    IsGettingAlertsucceed = true;
+
+    showSuccessMessage(" Alert Set !!");
+
+    alertBaseList.value = "";
+    alertTargetList.value = "";
+    alertRate.value = "";
+  }
+}
+
+function setAlertInterval(isValid) {
+  if (isValid) {
+    for (const key of intervalMap.keys()) {
+      stopInterval(key);
+    }
+
+    alerts.forEach((alertItem) => {
+      const currentRate = currencyRates.find((obj) => {
+        return obj.base === alertItem.base;
+      }).rates[alertItem.target];
+      const rowId = `${alertItem.base}-${alertItem.target}-row`;
+      const alertRow = document.getElementById(rowId);
+      const intervalId = `${alertItem.base}-${alertItem.target}-interval`;
+      if (alertItem.rate > currentRate) {
+        startInterval(
+          intervalId,
+          5000,
+          true,
+          alertItem.base,
+          alertItem.target,
+          alertItem.rate,
+          alertRow
+        );
+      } else if (alertItem.rate < currentRate) {
+        startInterval(
+          intervalId,
+          5000,
+          false,
+          alertItem.base,
+          alertItem.target,
+          alertItem.rate,
+          alertRow
+        );
+      }
+    });
+  }
+}
+
+function startInterval(
+  id,
+  time,
+  isIncrementalRate,
+  base,
+  target,
+  rate,
+  alertRow
+) {
+  const intervalId = setInterval(() => {
+    const newCurrentRate = currencyRates.find((obj) => {
+      return obj.base === base;
+    }).rates[target];
+    if (isIncrementalRate) {
+      if (newCurrentRate >= rate) {
```

Comment:
Isn't `isIncrementalRate` doing the same test?

Reply:
This I understood. My confusion comes from where you call this function:
```js
if (alertItem.rate > currentRate) {
  startInterval(
    intervalId,
    5000,
    true, // isIncrementalRate = true
    alertItem.base,
    alertItem.target,
    alertItem.rate,
    alertRow
  );
}
```

So you test  `alertItem.rate > currentRate` to define `isIncrementalRate`, and then you test again `newCurrentRate >= rate`.

Using two functions would probably help understanding your logic. Also, you can pass `alertItem` as an argument instead of passing each of its properties (`alertItem.base`,`alertItem.target`,...)